### ISSUE TITLE
Seed setting

### ISSUE THREAD

**Issue by liu15509348793** [ultralytics/ultralytics#15247]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, thank you for your work, I would like to ask if only the setting of the seed with the pytorch version above 2.0.0 will take effect, right?

**Comment by pderrenger** [comment]
@liu15509348793 thank you for your question. The seed setting should work with PyTorch versions both above and below 2.0.0. If you are experiencing issues with seed setting, please ensure you are using the latest version of the Ultralytics package. If the problem persists, feel free to provide more details so we can assist you further.

**Comment by liu15509348793** [comment]
def init_seeds(seed=0, deterministic=False):
    """Initialize random number generator (RNG) seeds https://pytorch.org/docs/stable/notes/randomness.html."""
    random.seed(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)
    torch.cuda.manual_seed_all(seed)  # for Multi-GPU, exception safe
    # torch.backends.cudnn.benchmark = True  # AutoBatch problem https://github.com/ultralytics/yolov5/issues/9287
    if deterministic:
        if TORCH_2_0:
            torch.use_deterministic_algorithms(True, warn_only=True)  # warn if deterministic is not possible
            torch.backends.cudnn.deterministic = True
            os.environ["CUBLAS_WORKSPACE_CONFIG"] = ":4096:8"
            os.environ["PYTHONHASHSEED"] = str(seed)
        else:
            LOGGER.warning("WARNING ⚠️ Upgrade to torch>=2.0.0 for deterministic training.")
    else:
        torch.use_deterministic_algorithms(False)
        torch.backends.cudnn.deterministic = False
Thank you for your reply, but I saw the above code in the torch_utils.py file, if the torch version is not 2.0.0 or above, the deterministic algorithms will not take effect, I look forward to you helping me answer my questions

**Comment by glenn-jocher** [comment]
@liu15509348793 you're correct. For deterministic algorithms to take effect, PyTorch 2.0.0 or above is required. If you're using an earlier version, deterministic behavior may not be fully supported.

**Comment by liu15509348793** [comment]
> @liu15509348793 you're correct. For deterministic algorithms to take effect, PyTorch 2.0.0 or above is required. If you're using an earlier version, deterministic behavior may not be fully supported.

ok,Thank you very much
